divi/src/script/standard.cpp

```diff
@@ -53,6 +54,17 @@ bool ExtractScriptPubKeyFormat(const CScript& scriptPubKey, txnouttype& typeRet,
 
         // Sender provides 2 pubkey hashes, 1-owner & 1-manager, receivers provides exactly 1 of the two signatures
         mTemplates.insert(std::make_pair(TX_VAULT, GetStakingVaultScriptTemplate() ));
+
+        // Sender provides two keys, a timelock and a hash value.  One of the
+        // keys can spend the output after the timelock, the other if they know
+        // a preimage to the hash value.
+        mTemplates.emplace(TX_HTLC, CScript () << OP_ANYHASH << OP_ANY << OP_EQUAL
+                                               << OP_IF
+                                               << OP_PUBKEYHASH
+                                               << OP_ELSE
+                                               << OP_ANY << OP_CHECKLOCKTIMEVERIFY << OP_DROP << OP_PUBKEYHASH
+                                               << OP_ENDIF
+                                               << OP_OVER << OP_HASH160 << OP_EQUALVERIFY << OP_CHECKSIG);
```

Comment:
The reason why `OP_CHECKLOCKTIMEVERIFY` does not automatically drop (i.e. consume the input) is because that would make it into a non-softfork.  By having it the way it is now, it is purely a restriction compared to the `OP_NOP2` it was before.

Reply:
That said, it would probably be fine to remove the small integer part entirely, as the values used here will be at least in the hundreds / thousands anyway (block heights for the time lock).

Reply:
Ah, I missed that bit. That makes more sense. Would it still make sense to allow for small integers? It's more complete but there's no practical reason why using a small integer would even be desirable.

Reply:
Yes exactly, for a true "anything", it makes sense to support small integers, that's why I did it like this in the current form.  For the purposes used here, it would be possible to simplify by removing the small-integer part, but I don't think it hurts, either.  So unless you prefer me to change it, I'd just keep it as it is.